src/agent/include/agent.hpp

```diff
@@ -1,18 +1,29 @@
 #pragma once
 
+#include <communicator.hpp>
 #include <task_manager.hpp>
 
 #include <queue>
 #include <string>
 
+
+using namespace communicator;
```

Comment:
It's generally not considered good practice to add `using namespace` declarations in header files. It can lead to namespace pollution and have some unintended consequences.

Reply:
Fixed